rlp/encbuffer.go

```diff
@@ -359,11 +359,16 @@ func (w *EncoderBuffer) ToBytes() []byte {
 }
 
 // AppendToBytes appends the encoded bytes to dst.
-func (w *EncoderBuffer) AppendToBytes(dst []byte) []byte {
+func (w *EncoderBuffer) AppendToBytes(dst *[]byte) {
 	size := w.buf.size()
-	out := append(dst, make([]byte, size)...)
-	w.buf.copyTo(out[len(dst):])
-	return out
+	tmp, length := *dst, len(*dst)
+	if cap(tmp) >= size+length {
+		tmp = tmp[:size+length]
+	} else {
+		tmp = append(tmp, make([]byte, size)...)
```

Comment:
If the rest capacity can't meets size requirement, then create a new slice and copy current data to it and recover the slice to dst.